File: dwaplatform/src/main/java/com/dwaplatform/android/transactions/ui/transactionDetail/ui/TransactionDetailUI.java
package com.dwaplatform.android.transactions.ui.transactionDetail.ui;

import android.content.Context;
import android.content.Intent;
import com.dwaplatform.android.transactions.models.TransactionItem;

public class TransactionDetailUI {
    protected static TransactionDetailUI instance;

    protected TransactionDetailViewComponent createTransactionDetailComponent(TransactionDetailContract.View view) {
        return DaggerTransactionDetailViewComponent.builder()
                .transactionDetailPresenterModule(new TransactionDetailPresenterModule(view))
                .build();
    }
    public TransactionDetailViewComponent buildTransactioDetailComponent(TransactionDetailContract.View view) {
        return instance.createTransactionDetailComponent(view);
    }

    public void start(Context context, TransactionItem item) {
        instance = this;
        Intent intent = new Intent(context, TransactionDetailActivity.class);
        intent.putExtra("transaction", item);
        context.startActivity(intent);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/detail/TransactionDetailUI.kt
package com.fintechplatform.ui.transactions.ui.detail

import android.content.Context
import android.content.Intent
import com.fintechplatform.ui.transactions.models.TransactionItem
import com.fintechplatform.ui.transactions.ui.detail.di.DaggerTransactionDetailViewComponent
import com.fintechplatform.ui.transactions.ui.detail.di.TransactionDetailPresenterModule
import com.fintechplatform.ui.transactions.ui.detail.di.TransactionDetailViewComponent

class TransactionDetailUI {

    fun start(context: Context, item: TransactionItem) {
        val intent = Intent(context, TransactionDetailActivity::class.java)
        intent.putExtra("transaction", item)
        context.startActivity(intent)
    }

    object Builder {

        fun buildTransactioDetailComponent(view: TransactionDetailContract.View): TransactionDetailViewComponent {
            return DaggerTransactionDetailViewComponent.builder()
                    .transactionDetailPresenterModule(TransactionDetailPresenterModule(view))
                    .build()
        }
    }
}